sn_node/src/node/messaging/handover.rs

```diff
@@ -28,20 +28,29 @@ impl MyNode {
     pub(crate) fn propose_handover_consensus(
         &mut self,
         sap_candidates: SapCandidate,
-    ) -> Result<Option<Cmd>> {
+    ) -> Result<Vec<Cmd>> {
+        let mut cmds = Vec::new();
         match &self.handover_voting {
```

Comment:
A bit unrelated, but the clones here seem a bit wasteful. Do you know if this would work?

```suggestion
        match &mut self.handover_voting {
```

This way we can mutate the handover state directly instead cloning and writing back the updated state.

Reply:
we cann't do it as within the match clause, the `self.broadcast_handover_vote_msg` and `self.broadcast_handover_decision` also requires mutable access to `self` which will get `locked` if we do the `as_mut` call.